### PR TITLE
pull fixes

### PR DESCRIPTION
**Pull request by snirazran**

### CODE REVIEW

File: apps/frontend/src/axios/axios.ts [hunk 1]
```diff
@@ -1,17 +1,28 @@
...
+  baseURL: config.baseURL,
   timeout: 5000,
   headers: {
     'Content-Type': 'application/json',
   },
 });
 
-const getAccessToken = () => {
+const getAccessToken = (): string | null => {
   const userData = localStorage.getItem('user');
```
Comment (yarinsa) [inline apps/frontend/src/axios/axios.ts:23]: Move to a new user.tsx file , export. { get, set }. Don't use "localStorage" in this file. The logic of where the user is shouldn't be in "getAccessToken" function

File: apps/frontend/src/axios/axios.ts [hunk 2]
```diff
@@ -1,17 +1,28 @@
...
-  const user = JSON.parse(userData);
-  return user.token;
+  const user: User | null = userData ? JSON.parse(userData) : null;
```
Comment (yarinsa) [inline apps/frontend/src/axios/axios.ts:24]: Move to user.tsx (get) , create a type guard so type will be inferred. (Remove the manual typing of User)

File: apps/frontend/src/axios/axios.ts [hunk 3]
```diff
@@ -1,17 +1,28 @@
...
+  return user ? user.token : null;
```
Comment (yarinsa) [inline apps/frontend/src/axios/axios.ts:25]: User?.token ?? Null

File: apps/frontend/src/config.ts [hunk 4]
```diff
@@ -0,0 +1,14 @@
+const config = {
+  baseURL: 'http://localhost:5173',
+  firebaseConfig: {
+    apiKey: (import.meta.env.VITE_API_KEY as string) ?? '',
```
Comment (yarinsa) [inline apps/frontend/src/config.ts:4]: Now if it's undefined we hide the bug with empty string. How about create a get function that throws error if something is undefined.

File: apps/frontend/src/features/users/useGetAllUsers.tsx [hunk 5]
```diff
@@ -4,7 +4,15 @@ import userService from './userService';
 import useSWR from 'swr';
 import useLocalStorage from '../../utils/useLocalStorage';
 
-type User = { name: string };
+type User = {
```
Comment (yarinsa) [inline apps/frontend/src/features/users/useGetAllUsers.tsx:7]: Move to user.tsx

File: apps/frontend/src/firebase.js [hunk 6]
```diff
@@ -1,8 +1,8 @@
 import { initializeApp } from 'firebase/app';
 import { getStorage } from 'firebase/storage';
-import { firebaseConfig } from './config';
+import config from './config';
 
-const app = initializeApp(firebaseConfig);
+const app = initializeApp(config.firebaseConfig);
 const storage = getStorage(app, import.meta.env.VITE_FIREBASE_STORAGE);
```
Comment (yarinsa) [inline apps/frontend/src/firebase.js:6]: Use the the config instead of Env var

**Review by yarinsa — COMMENTED** [review]
Looks good, left few more comments for cleaner code

File: apps/frontend/src/axios/axios.ts (hunk 1 above)
Comment (snirazran) [reply apps/frontend/src/axios/axios.ts:23]: are you talking about a user class with get and set functions or just a function like the one i made that gets the user from local storage

Comment (yarinsa) [reply apps/frontend/src/axios/axios.ts:23]: After transpilation a file :
file with class = object the exports object with function
file with named exports = object that exports function

(There is no benefit for class here, it's more OOP but we like functional programming)

**Pull request merged**